packages/workflow-apps-defaults/src/index.js

```diff
@@ -1,69 +1,110 @@
 /* eslint-env node */
 /* eslint-disable no-console */
-
 import { baseFolder } from 'shared/env';
 import { join } from 'path';
+import defaultBrowser from 'default-browser';
 
-const defaultApps = ['Terminal', 'Browser', 'TextEditor'];
-
-let defaults = {};
+function getUserDefaults() {
+  try {
+    const path = join(baseFolder, 'apps', 'defaults');
+    const defaultModule = require(path);
 
-if (process.browser) {
-  defaults = require('workflow-apps-html').defaults;
-} else {
-  const platformDefaults = (() => {
-    switch (process.platform) {
-      case 'darwin':
-        return {
-          Terminal: require('workflow-app-iterm').ITerm,
-          Browser: require('workflow-app-safari').Safari,
-          TextEditor: require('workflow-app-textedit').TextEdit,
-        };
-      case 'win32':
-        return {
-          Terminal: require('workflow-app-powershell').PowerShell,
-          Browser: require('workflow-app-chrome').Chrome,
-          TextEditor: require('workflow-app-notepad').Notepad,
-        };
-      case 'linux':
-        return {
-          Terminal: require('workflow-app-xterm').XTerm,
-          Browser: require('workflow-app-chrome').Chrome,
-          TextEditor: require('workflow-app-atom').Atom,
-        };
-      default:
-        console.log(`Platform '${process.platform}' not supported`);
-        console.log(
-          'Look for an issue for your platform here: https://github.com/havardh/workflow/issues'
-        );
-        process.exit(0);
-        break;
+    return defaultModule.default || defaultModule;
+  } catch (error) {
+    if (error.code !== 'MODULE_NOT_FOUND') {
+      throw error;
+    } else {
+      return {};
     }
-    throw new Error('not reachable');
-  })();
+  }
+}
 
-  const userDefaults = (() => {
-    try {
-      const path = join(baseFolder, 'apps', 'defaults');
-      const { defaults } = require(path);
+function throwUnknownPlatform() {
+  console.log(`Platform '${process.platform}' not supported`);
+  console.log(
+    'Look for an issue for your platform here: https://github.com/havardh/workflow/issues'
+  );
+  process.exit(0);
+}
 
-      return defaults || {};
-    } catch (error) {
-      if (error.code !== 'MODULE_NOT_FOUND') {
-        throw error;
-      } else {
-        return {};
-      }
-    }
-  })();
+async function requireBrowser(browserName, fallback) {
+  try {
+    return require(`workflow-app-${browserName}`);
```

Comment:
Need to handle the named export here.

Reply:
What about:

1. If there is excacly one named export, use it.
2. If there is a named export exactly matching `pascalCase(browserName)` use it.
3. Throw an exception about usage in `apps/defaults` where the actual name can be set explicitly.

Reply:
I was thinking just requiring 2 🙃

Reply:
If so we would have a direct dependency on the `default-browser` package which we do not control, and if it returns a different name for the same application on different platforms. Then we are out of luck...

Reply:
We already do since we pass it directly to require. The only requirement here is that is new is that the export matches the package name.